kura/org.eclipse.kura.http.server.manager/src/main/java/org/eclipse/kura/http/server/manager/HttpService.java

```diff
@@ -97,22 +101,33 @@ private synchronized void restartHttpService() {
     }
 
     private synchronized void activateHttpService() {
-        try {
-            logger.info("starting Jetty instance...");
-            this.jettyServerHolder = new JettyServerHolder(this.options, Optional.ofNullable(this.keystoreService),
-                    this.dispatcherServlet, this.eventListener);
-            logger.info("starting Jetty instance...done");
-        } catch (final Exception e) {
-            logger.error("Could not start Jetty Web server", e);
-        }
+        this.executorService.submit(() -> {
+            try {
+                logger.info("starting Jetty instance...");
+                this.jettyServerHolder = new JettyServerHolder(this.options, Optional.ofNullable(this.keystoreService),
```

Comment:
To avoid race conditions we can try passing an immutable copy of the current options instead of referencing this.options directly. The activateHttpService method can be modified to take the options as argument and having the activate() and update() method passing the new options to this one.